raster/r.out.gdal/tests/test_r_out_gdal.py

```diff
@@ -0,0 +1,99 @@
+import grass.script as gs
+import pytest
+from grass.tools import Tools
+
+FORMAT_DICT = {
+    "COG": "tif",
+    "GTiff": "tif",
+    "GPKG": "gpkg",
+    "VRT": "vrt",
+    "netCDF": "nc",
+    "ENVI": "img",
+}
+
+
+@pytest.fixture
+def simple_raster_map(tmp_path_factory):
+    """Fixture to create a basic GRASS environment with a simple raster map containing attributes.
+
+    Set up a temporary GRASS project and region and create a raster map
+    using r.mapcalc.
+
+    Yields:
+        tuple: (map name, GRASS session handle, tmp_path)
+
+    """
+    tmp_path = tmp_path_factory.mktemp("r_out_gdal_project")
+    project = tmp_path / "grassdata"
+    gs.create_project(project, epsg=4326)
+
+    with gs.setup.init(project) as session:
+        tools = Tools(session=session)
+        tools.g_region(n=10, s=0, e=10, w=0, res=1)
+
+        tools.r_mapcalc(
+            expression="test_raster=row() * col()",
+        )
+
+        yield "test_raster", session, tmp_path
+
+
+@pytest.mark.parametrize("file_format", list(FORMAT_DICT.keys()))
+def test_basic_cog_export(simple_raster_map, file_format):
+    """Test basic export of various formats using 'r.out.gdal'.
+
+    Verifies:
+    - Export of common format works with defaults.
+    """
+    mapname, session, tmp_path = simple_raster_map
+    tools = Tools(session=session)
+    suffix = FORMAT_DICT[file_format]
+    output_file = tmp_path / f"{mapname}.{suffix}"
+    tools.r_out_gdal(
+        input=mapname,
+        output=str(output_file),
+        format=file_format,
+    )
+    assert output_file.exists(), f"{mapname} not exportd to {output_file}"
+
+
+def test_compressed_gtiff_export(simple_raster_map):
+    """Test export of compressed GeoTiff using 'r.out.gdal'.
+
+    Verifies:
+    - Export of common format works.
+    """
+    mapname, session, tmp_path = simple_raster_map
+    tools = Tools(session=session)
+
+    output_file = tmp_path / f"{mapname}.tif"
+    tools.r_out_gdal(
+        input=mapname,
+        output=str(output_file),
+        format="GTiff",
+        createopt="COMPRESS=LZW",
+    )
+    assert output_file.exists(), f"{mapname} not exportd to {output_file}"
+
+
+def test_fast_gtiff_export(simple_raster_map):
+    """Test fast export of compressed GeoTiff using 'r.out.gdal'.
+
+    Verifies:
+    - Fast export of works if type, nodata and -f are given.
+    """
+    mapname, session, tmp_path = simple_raster_map
+    tools = Tools(session=session)
+
+    output_file = tmp_path / f"{mapname}.tif"
+    tools.r_out_gdal(
+        flags="f",
+        input=mapname,
+        output=str(output_file),
+        format="GTiff",
+        type="Byte",
+        nodata=255,
+        createopt="COMPRESS=LZW",
+    )
+    assert output_file.exists(), f"{mapname} not exportd to {output_file}"
+    # Here I would like to assert "Checking GDAL data type and nodata value" not in std_err
```

Comment:
I cannot figure out how to capture the std_err with `Tools`... Setting the _capture_stderr_ parameter in Tools to True does not seem to be enough... Any hints are moste welcome...

Reply:
If it wasn't of the grass-specific API discussed here, the pytest way is with the fixture like capsys/capfd, and reading it, like shown and explained in https://docs.pytest.org/en/stable/how-to/capture-stdout-stderr.html#accessing-captured-output-from-a-test-function

Reply:
> ```python
> result = tools.r_out_gdal(...)
> assert "Checking GDAL data type and nodata value" not in result.stderr
> ```
> 
> Is the the above not working for your or is it not clear that that's the way?

I had/added that kind of code in the test but tools seemed to only return a None-type (see: https://github.com/OSGeo/grass/actions/runs/18057685017/job/51389745313). Strange enough, as it seemed to work fine in other tests. Not sure what might cause the issue...